internal/cli/internal/command/machine/clone.go

```diff
@@ -0,0 +1,154 @@
+package machine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/superfly/flyctl/api"
+
+	"github.com/superfly/flyctl/cmd/presenters"
+	"github.com/superfly/flyctl/helpers"
+
+	"github.com/superfly/flyctl/pkg/iostreams"
+	"github.com/superfly/flyctl/pkg/logs"
+
+	"github.com/superfly/flyctl/internal/cli/internal/app"
+	"github.com/superfly/flyctl/internal/cli/internal/command"
+	"github.com/superfly/flyctl/internal/cli/internal/config"
+	"github.com/superfly/flyctl/internal/cli/internal/flag"
+	"github.com/superfly/flyctl/internal/cli/internal/prompt"
+	"github.com/superfly/flyctl/internal/client"
+	"github.com/superfly/flyctl/internal/logger"
+)
+
+func newClone() *cobra.Command {
+	const (
+		short = "Clones a Fly Machine"
+		long  = short + "\n"
+
+		usage = "clone <id>"
+	)
+
+	cmd := command.New(usage, short, long, runMachineClone,
+		command.RequireSession,
+		command.RequireAppName,
+	)
+
+	cmd.Args = cobra.ExactArgs(1)
+
+	flag.Add(
+		cmd,
+		flag.App(),
+		flag.AppConfig(),
+		flag.String{
+			Name:        "region",
+			Shorthand:   "r",
+			Description: "Target region",
+		},
+		flag.String{
+			Name:        "name",
```

Comment:
Should this be called ID? I'm not clear on what the difference is between the machine alloc ID and its name.

Reply:
I can't tell either

Reply:
I think since the goal of this PR is to port, we can keep this convention and update later